packages/headless/src/app/search-engine/search-engine.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  NoopPostprocessFacetSearchResponseMiddleware,
+  NoopPostprocessQuerySuggestResponseMiddleware,
+  NoopPostprocessSearchResponseMiddleware,
+} from '../../api/search/search-api-client-middleware';
+import {buildEngine, EngineOptions, ExternalEngineOptions} from '../engine';
+import {Engine} from '../headless-engine';
+import {buildLogger} from '../logger';
+import {buildThunkExtraArguments} from '../thunk-extra-arguments';
+import {Logger} from 'pino';
+import {NoopPreprocessRequest} from '../../api/preprocess-request';
+import {NoopPreprocessRequestMiddleware} from '../../api/platform-client';
+import {debug, pipeline, search, searchHub} from '../reducers';
+import {StateFromReducersMapObject} from 'redux';
+import {updateSearchConfiguration} from '../../features/configuration/configuration-actions';
+import {
+  SearchEngineConfiguration,
+  searchEngineConfigurationSchema,
+} from './search-engine-configuration-options';
+import {executeSearch} from '../../features/search/search-actions';
+import {logInterfaceLoad} from '../../features/analytics/analytics-actions';
+import {firstSearchExecutedSelector} from '../../features/search/search-selectors';
+import {SearchAppState} from '../../state/search-app-state';
+
+const searchEngineReducers = {debug, pipeline, searchHub, search};
+type SearchEngineReducers = typeof searchEngineReducers;
+type SearchEngineState = StateFromReducersMapObject<SearchEngineReducers> &
+  Partial<SearchAppState>;
+
+export interface SearchEngine extends Engine<SearchEngineState> {
+  executeFirstSearch(): void;
+}
+
+export interface SearchEngineOptions
+  extends ExternalEngineOptions<SearchEngineState> {
+  /**
+   * The search engine configuration options.
+   */
+  configuration: SearchEngineConfiguration;
+}
+
+export function buildSearchEngine(options: SearchEngineOptions): SearchEngine {
+  const logger = buildLogger(options.loggerOptions);
+  validateConfiguration(options.configuration, logger);
+
+  const ref = {
+    renewAccessToken: () => Promise.resolve(''),
+  };
+
+  const searchAPIClient = createSearchAPIClient(
+    options.configuration,
+    logger,
+    ref
+  );
+
+  const thunkArguments = {
+    ...buildThunkExtraArguments(options.configuration, logger),
+    searchAPIClient,
+  };
+
+  const augmentedOptions: EngineOptions<SearchEngineReducers> = {
+    ...options,
+    reducers: searchEngineReducers,
+  };
+
+  const engine = buildEngine(augmentedOptions, thunkArguments);
+  ref.renewAccessToken = engine.renewAccessToken;
+
+  const {search} = options.configuration;
+
+  if (search) {
+    engine.dispatch(updateSearchConfiguration(search));
+  }
+
+  return {
+    ...engine,
+
+    get state() {
+      return engine.state;
+    },
+
+    executeFirstSearch() {
+      const firstSearchExecuted = firstSearchExecutedSelector(engine.state);
+
+      if (firstSearchExecuted) {
+        return;
+      }
+
+      const action = executeSearch(logInterfaceLoad());
```

Comment:
Hum, it's not always going to be this event.

The first search can be done for other reasons than an interface load.
For example, if you get redirected from a standalone search box.

So I would suggest a parameter, (with default value being logInterfaceLoad)

Reply:
Should we allow any search analytics function? Or maybe we can scope it to just interfaceLoad and standalone search box causes? e.g.

```
type Cause = 'interfaceLoad' | 'searchFromLink' | 'omniboxFromLink' 
executeFirstSearch(cause: Cause = 'interfaceLoad') {}
```

If someone needs a different thing, a dev could still dispatch the action themselves. Headless would keep the principle of "more effort to do the wrong thing".

I [created a jira](https://coveord.atlassian.net/browse/KIT-702) to add the functionality, since the analytics event action wrappers for a standalone search box are not coded.